core-server/src/main/java/org/glassfish/jersey/server/internal/scanning/PackageNamesScanner.java

```diff
@@ -107,15 +110,19 @@ public PackageNamesScanner(final ClassLoader classLoader, final String[] package
         this.classloader = classLoader;
 
         this.finderFactories = new HashMap<>();
-        add(new JarZipSchemeResourceFinderFactory());
-        add(new FileSchemeResourceFinderFactory());
-        add(new VfsSchemeResourceFinderFactory());
-        add(new BundleSchemeResourceFinderFactory());
 
-        // TODO - Services?
-        // for (UriSchemeResourceFinderFactory s : ServiceFinder.find(UriSchemeResourceFinderFactory.class)) {
-        //     add(s);
-        // }
+        ServiceFinder<UriSchemeResourceFinderFactory> servicesFound = ServiceFinder.find(UriSchemeResourceFinderFactory.class);
+
+        // TODO: ServiceFinder should return an Optional
+        if (servicesFound.toArray().length == 0) {
```

Comment:
I would write the whole construction shorter, like:

for (UriSchemeResourceFinderFactory s : ((servicesFound.toArray().length == 0) ? 
                                                                defaultFinderFactories : servicesFound)) {
            add(s);
        }

but here could be objections from people who likes IFs...

Reply:
Either way for me it's fine. Personally, i think the code will be more readable with the `If` Statement there.

Reply:
Do you want me to apply your suggestion?